flink-libraries/flink-table/src/main/scala/org/apache/flink/table/calcite/RelTimeIndicatorConverter.scala

```diff
@@ -415,7 +406,13 @@ class RexTimeIndicatorMaterializer(
       case _ =>
         updatedCall.getOperands.map { o =>
           if (isTimeIndicatorType(o.getType)) {
-            rexBuilder.makeCall(TimeMaterializationSqlFunction, o)
+            if (isRowtimeIndicatorType(o.getType)) {
```

Comment:
Not sure. The duplicated code is spread over 2 classes and differently parameterized. I tried to add a companion object to `RexTimeIndicatorMaterializer` and the method but this did not really improve the code, IMO.